Scripts/Mac/BuildGVFSForMac.sh

```diff
@@ -45,4 +45,13 @@ echo 'Copying native binaries to Publish directory...'
 cp $VFS_OUTPUTDIR/GVFS.Native.Mac/Build/Products/$CONFIGURATION/GVFS.ReadObjectHook $VFS_PUBLISHDIR || exit 1
 cp $VFS_OUTPUTDIR/GVFS.Native.Mac/Build/Products/$CONFIGURATION/GVFS.VirtualFileSystemHook $VFS_PUBLISHDIR || exit 1
 
+echo 'Copying Git installer to Publish directory...'
+GITPUBLISH=$VFS_PUBLISHDIR/Git
+if [[ -d $GITPUBLISH ]] ; then
+  rm -rf $GITPUBLISH
+fi
+mkdir $GITPUBLISH
+cp $GITPATH $GITPUBLISH
```

Comment:
Why is it necessary to actually copy? Couldn't we just update the init script to set the `GITPATH` variable appropriately?

Two concerns with this:
* build+publish+unit tests is already getting to be too long. Adding a copy of a large package is just going to add to that. Maybe do the copy only if it's not already there?
* The `VFS_PUBLISHDIR` currently contains just the outputs of `dotnet publish GVFS.sln`. With this change, that publish folder will now contain different contents if I build+publish from VS or build+publish with this script. So if we do need to copy, I'd say copy the Git package to a parallel folder, not inside the same publish folder.

Thoughts?

Reply:
> Couldn't we just update the init script to set the GITPATH variable appropriately?

I'm not a fan of that, since it means we need to sweep up the packages (which I view as build _input_) and pass it between jobs. I think that if we need something in a package available to end users, the build should promote it to an output folder.

> build+publish+unit tests is already getting to be too long. Adding a copy of a large package is just going to add to that. Maybe do the copy only if it's not already there?

Makes sense to me. I'll make that change.

> So if we do need to copy, I'd say copy the Git package to a parallel folder, not inside the same publish folder.

Also makes sense. Thanks!

Reply:
To add one more element here - today I can do a clean build+publish from VS, then run the prep script, and then run the functional tests.

With this change, we would now require that the build happen via this build script, at least once. It would be nice to keep the ability to build from VS if possible.

Reply:
@sanoursa and I agreed offline that, unfortunately, clean build+publish from VS already doesn't work, and the build script is already a prereq. I'm adding to the problem here. We'll move more of this into the build phase itself in subsequent refactorings, but it's deferred while we work on moving to yaml pipelines.

Reply:
Sounds good to me. Note that I plan to spend some time next week digging into getting our build working well in VSMac (or at least understanding why things don't work well today).